app.territoiresentransitions.react/src/app/pages/collectivite/PlansActions/ToutesLesFichesAction/MenuFiltresToutesLesFichesAction.tsx

```diff
@@ -59,374 +47,457 @@ const MenuFiltresToutesLesFichesAction = ({
 
   const debutPeriodeRef = useRef<HTMLInputElement>(null);
   const finPeriodeRef = useRef<HTMLInputElement>(null);
+  const { register, watch } = useForm({ defaultValues: filters });
+  const { noPriorite, noTag, isBelongsToSeveralPlans }: Filtres = watch();
+
+  useEffect(() => {
+    let newFilters: Filtres = {
+      ...filters,
+      ...{
+        noPriorite: noPriorite ? noPriorite : undefined,
+        noTag: noTag ? noTag : undefined,
+        isBelongsToSeveralPlans: isBelongsToSeveralPlans
+          ? isBelongsToSeveralPlans
+          : undefined,
+      },
+    };
+    newFilters = removeFalsyElementFromFilters(newFilters);
+    setFilters(newFilters);
+  }, [noPriorite, noTag, isBelongsToSeveralPlans]);
 
   return (
     <div className="w-96 md:w-[48rem] p-4 lg:p-8">
-      <FormSection
-        title={title}
-        className="!grid-cols-1 md:!grid-cols-2 gap-x-8"
-      >
-        <div className="*:mb-4 first:!mb-0">
-          <Field title="Plans d'action">
-            <PlansActionDropdown
-              values={filters.planActionIds}
-              onChange={({ plans }) => {
-                const { planActionIds, ...rest } = filters;
-                setFilters({
-                  ...rest,
-                  ...(plans ? { planActionIds: plans } : {}),
-                });
-              }}
-            />
-          </Field>
+      <form>
+        <FormSection
+          title={title}
+          className="!grid-cols-1 md:!grid-cols-2 gap-x-8"
+        >
+          <div className="*:mb-4 first:!mb-0">
+            <Field title="Plans d'action">
+              <PlansActionDropdown
+                values={filters.planActionIds}
+                onChange={({ plans }) => {
+                  const { planActionIds, ...rest } = filters;
+                  setFilters({
+                    ...rest,
+                    ...(plans ? { planActionIds: plans } : {}),
+                  });
+                }}
+              />
```

Comment:
Avec react-hook-form only, on aurait ceci:

```suggestion
import { Controller } from 'react-hook-form';

<Controller
  name="planActionIds"
  control={control}
  render={({ field }) => (
    <PlansActionDropdown
      values={field.value}
      onChange={({ plans }) => {
        field.onChange(plans || null);
      }}
    />
  )}
/>
```
```